File: week07/20/pr1_Pascal's_triangle copy.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* getRow(int rowIndex, int* returnSize) {
        *returnSize = rowIndex+1;
        int* arr = (int*)malloc((*returnSize)*sizeof(int));
        for(int i=0;i<*returnSize;i++){
            arr[i] = 1;
        }
        if(*returnSize==0 || *returnSize==1) return arr;
        else{
            int* temp = getRow(rowIndex-1,returnSize);
            for(int j=1;j<*returnSize;j++){
                arr[j] = temp[j-1]+temp[j];
            }
            free(temp);
            *returnSize = rowIndex+1;
            return arr;}
}

File: week07/20/pr1_Pascal's_triangle copy.py
class Solution:
    def getRow(self, rowIndex: int) -> list[int]:
        if rowIndex == 0 : return [1]
        elif rowIndex == 1 : return [1,1]
        
        temp = self.getRow(rowIndex-1)
        ans = [1]*(rowIndex+1)
        for i in range(1,rowIndex):
            ans[i]=temp[i-1]+temp[i]
        
        return ans